### PR TITLE
@joeyAghion => [Messaging] Display unread indicator based on Metaphysics response

### PR DESCRIPTION
**Pull request by mzikherman**
screen shot 2017-08-11 at 10 31 08 am

Once https://github.com/artsy/metaphysics/pull/699 is merged, I'll update the temp code here.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "tslint": "tslint",
     "sync-colors": "cd externals/elan && git fetch && git checkout origin/master && cp components/lib/variables/colors.json ../../data",
     "sync-externals": "npm run-script sync-schema && npm run-script sync-colors",
-    "sync-schema": "cd externals/metaphysics && git fetch && git checkout origin/master && yarn install && npm run dump-schema -- ../../data && rm -rf node_modules",
+    "sync-schema": "cd externals/metaphysics && git fetch && git checkout origin/desired_unread_api && yarn install && npm run dump-schema -- ../../data && rm -rf node_modules",
```
Comment (mzikherman) [inline package.json]: Temp code (was using this to dump the schema from the PR'ed branch).

File: src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx [hunk 2]
```diff
@@ -169,7 +169,7 @@ export class ConversationSnippet extends React.Component<Props, any> {
                   <MetadataText>
                     {date}
                   </MetadataText>
-                  <UnreadIndicator />
+                  {conversation.is_last_message_to_user && !conversation.last_message_open && <UnreadIndicator />}
```
Comment (mzikherman) [inline src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx]: This is the one line to make the unread indicator accurate (based on the Metaphysics data).

File: src/lib/relay/config.tsx [hunk 3]
```diff
@@ -13,6 +13,8 @@ if (Emission && Emission.useStagingEnvironment) {
   gravityURL = "https://api.artsy.net"
 }
 
+metaphysicsURL = "http://localhost:5001"
```
Comment (mzikherman) [inline src/lib/relay/config.tsx]: Other temp code (pointing to Metaphysics locally).

File: package.json (hunk 1 above)
Comment (alloy) [inline package.json]: FYI The dump-schema script lives in the MP repo, so you can always just run it from your local dev checkout and specify the path to your local Emission checkout.

File: src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx (hunk 2 above)
Comment (alloy) [inline src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx]: Why is the `conversation.is_last_message_to_user` part necessary?

Comment (mzikherman) [inline src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx]: B/c the unread indicator is supposed to indicate if there's new content for the user. So if the last message in a thread is the user sending something to the gallery, we wouldn't want to show an unread indicator next to that thread.

Comment (alloy) [inline src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx]: Ok, I see. But in that case wouldn’t it be weird for `conversation.last_message_open` to return `false`?

**Comment by joeyAghion** [comment]
https://github.com/artsy/metaphysics/pull/699 is merged. Update away!

File: src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx (hunk 2 above)
Comment (mzikherman) [inline src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx]: Well, that's a string/timestamp of the open, so checking the initial boolean should suffice.

Aka, it doesn't matter if `last_message_open` is `null` or not. So if the last message is to the user, and it hasn't been opened yet- show the indicator. Any other cases- we wouldn't want to show it.

Comment (alloy) [inline src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx]: Ok, yeah I see 👍

**Comment by mzikherman** [comment]
📗 shd be good to go!

**Pull request merged**